golem/managers/agreement/default.py

```diff
@@ -20,8 +21,16 @@ def __init__(
         self._get_draft_proposal = get_draft_proposal
         self._event_bus = golem.event_bus
 
+        self._agreements: List[Agreement] = []
```

Comment:
This should be more managed. Termination of agreement should remove it from `self._agreements`.